### PR TITLE
Add Concat processor and test helpers

### PR DESCRIPTION
**Pull request by martingagnon**
## Description
Add `Concat` processor (also `StartWith` extension function)

Add many tests helpers to validate subscriptions.

## How Has This Been Tested?
See added tests 

## Types of changes
- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: documentation/PUBLISHERS.md [hunk 1]
```diff
@@ -207,6 +207,21 @@ uppercasePublisher.subscribe(...)
 ```
 In this case,  when fooPublisher emit a new value, the maps will only be executed once.
 
+
+#### Concat processor
+Emits the values of the first publisher until it completes. Then emits the values for the nextPublisher.
+```kotlin
+val firstPublisher = Publishers.behaviorSubject("i")
+val nextPublisher = Publishers.behaviorSubject("concat")
```
Comment (jdtremblay) [inline documentation/PUBLISHERS.md]: Il sert à quoi `nextPublisher` dans ton exemple?

Comment (martingagnon) [reply documentation/PUBLISHERS.md]: Damn, my mistake

Comment (martingagnon) [reply documentation/PUBLISHERS.md]: Fixed!

File: streams/src/commonTest/kotlin/com/mirego/trikot/streams/reactive/processors/ConcatProcessorTests.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        secondPublisher.value = "1"
+        firstPublisher.value = "b"
+        firstPublisher.complete()
+        assertEquals(listOf("a", "b", "1"), accumulator.values)
+        assertEquals(false, accumulator.completed)
+        assertEquals(null, accumulator.error)
+    }
+
+    @Test
+    fun valuesFromSecondPublisherAreBufferedUntilFirstPublisherCompleted() {
```
Comment (mbaron) [inline streams/src/commonTest/kotlin/com/mirego/trikot/streams/reactive/processors/ConcatProcessorTests.kt]: De we really want to accumulate and buffer values on the second "Hot" observable?

**Pull request merged**